### PR TITLE
HAAR-881 Add endpoint to search prisoner users by firstname and lastname

### PR DESCRIPTION
**Pull request by karenmillermoj**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/config/AuthWebClientConfiguration.kt [hunk 1]
```diff
@@ -19,9 +19,16 @@ class AuthWebClientConfiguration(appContext: ApplicationContext) : AbstractWebCl
   fun authWebClient(builder: Builder, authorizedClientManager: OAuth2AuthorizedClientManager): WebClient =
     getWebClient(builder, authorizedClientManager)
 
+  @Bean
+  fun authUserWebClient(builder: Builder, authorizedClientManager: OAuth2AuthorizedClientManager) =
```
[the remark this answers is not in the document]
Comment (karenmillermoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/config/AuthWebClientConfiguration.kt:23]: Very odd why intelliJ didn't show the error - will update

File: src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/resource/nomis/UserController.kt [hunk 2]
```diff
@@ -82,6 +88,55 @@ class UserController(
...
+        description = "OK"
+      ),
+      ApiResponse(
+        responseCode = "401",
+        description = "Unauthorized.",
+        content = [
+          Content(
+            mediaType = "application/json",
+            schema = Schema(implementation = ErrorResponse::class)
+          )
```
[the remark this answers is not in the document]
Comment (karenmillermoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/resource/nomis/UserController.kt:111]: There seems to be a bit of a mishmash with these.  We should probably create a ticket to standardise them

File: src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/resource/nomis/UserController.kt [hunk 3]
```diff
@@ -82,6 +88,55 @@ class UserController(
...
+  ): List<PrisonUser> = nomisUserService.findUsersByFirstAndLastName(firstName, lastName)
+    .map {
+      PrisonUser(
+        username = it.username,
+        staffId = it.userId.toLongOrNull(),
+        email = it.email,
+        verified = it.verified,
+        firstName = WordUtils.capitalizeFully(it.firstName),
+        lastName = WordUtils.capitalizeFully(it.lastName),
+        name = WordUtils.capitalizeFully("${it.firstName} ${it.lastName}"),
```
[the remark this answers is not in the document]
Comment (karenmillermoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/resource/nomis/UserController.kt:136]: yes - good point

**Review by SimonMitchellMOJ — APPROVED** [review]
LGTM

File: src/test/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/resource/nomis/UserControllerIntTest.kt [hunk 4]
```diff
@@ -436,4 +437,57 @@ class UserControllerIntTest : IntegrationTestBase() {
...
+              "lastName" to "Last",
+              "name" to "First Last",
+              "activeCaseLoadId" to "MDI",
+            )
+          )
+        }
+    }
+
+    @Test
+    fun `Prison user not found`() {
```
[the remark this answers is not in the document]
Comment (karenmillermoj) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/resource/nomis/UserControllerIntTest.kt:481]: 404 is not an issue - it returns an empty [] if nothing matches

File: src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/resource/nomis/UserController.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (olatomoloju) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/resource/nomis/UserController.kt:111]: OK

File: src/test/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/resource/nomis/UserControllerIntTest.kt (hunk 4 above)
[the remark this answers is not in the document]
Comment (olatomoloju) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/resource/nomis/UserControllerIntTest.kt:481]: OK

File: src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/resource/nomis/UserController.kt (hunk 3 above)
[the remark this answers is not in the document]
Comment (olatomoloju) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/resource/nomis/UserController.kt:136]: U probably didn't get my suggestion. But its fine.

[the remark this answers is not in the document]
Comment (karenmillermoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/resource/nomis/UserController.kt:136]: I saw it and thought it was a good idea but then realised firstName and lastName are part of the constructor, not variables (please see my reply)

**Pull request merged**